skyportal/handlers/api/source.py

```diff
@@ -243,17 +254,26 @@ def get(self, obj_id=None):
             )
 
             return self.success(data=source_info)
-        q = Obj.query.filter(
-            Obj.id.in_(
-                DBSession()
-                .query(Source.obj_id)
-                .filter(
-                    Source.group_id.in_(
-                        [g.id for g in self.current_user.accessible_groups]
-                    )
+
+        # Fetch multiple sources
+        user_group_ids = [g.id for g in self.current_user.accessible_groups]
+        q = (
+            DBSession()
+            .query(Obj)
+            .join(Source)
+            .filter(
+                Source.group_id.in_(  # only give sources the user has access to
+                    [g.id for g in self.current_user.accessible_groups]
```

Comment:
```suggestion
                    user_group_ids
```

Reply:
Yes, also replaced a similar line above that (moved the definition of `user_group_ids` up to the beginning.